aiohttp/web_urldispatcher.py

```diff
@@ -230,8 +231,24 @@ def set_current_app(self, app):
     def freeze(self):
         self._frozen = True
 
+    @property
+    def variable_maps(self):
+        return tuple(reversed(self._variables.maps))
+
+    def add_variables(self, match_dict):
+        self._variables.maps.append(match_dict)
+
+    def __getitem__(self, key):
+        return self._variables[key]
+
+    def __iter__(self):
+        return self._variables.__iter__()
+
+    def __len__(self):
+        return self._variables.__len__()
+
     def __repr__(self):
-        return "<MatchInfo {}: {}>".format(super().__repr__(), self._route)
+        return "<MatchInfo {}: {}>".format(dict(self).__repr__(), self._route)
```

Comment:
`repr(self._variables)` instead of `repr(dict(self))` is better I believe. There is no reason to hide chained variables.

Reply:
This was to make existing test cases that compares the `repr` result to pass. Then I will change those test cases to match with `ChainMap`'s `repr` result.